[MNG-7377] - ignore .vscode directories

jmhodges opened the pull request:
vscode creates directories with configurations in git repositories. This
is usually fine, but triggers the `apache-rat` license alarms.

To solve the alarm and avoid future issues from vscode contributors, we
exclude `.vscode/` from rat and add it to .gitignore.

Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MNG) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MNG-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MNG-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [ ] You have run the [Core IT][core-its] successfully. (nope, see https://issues.apache.org/jira/browse/MNG-7378)

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

[core-its]: https://maven.apache.org/core-its/core-it-suite/

pzygielo commented on .gitignore:
```diff
@@ -15,3 +15,4 @@ out/
 .java-version
 .checkstyle
 .factorypath
+.vscode/
```
This should go to global git config, not here.

pzygielo commented on pom.xml:
```diff
@@ -579,6 +579,7 @@ under the License.
               <exclude>src/main/appended-resources/licenses/MIT-slf4j-api-1.7.30.txt</exclude>
               <exclude>src/main/appended-resources/licenses/EPL-1.0.txt</exclude>
               <exclude>src/main/appended-resources/licenses/unrecognized-javax.annotation-api-1.3.2.txt</exclude>
+              <exclude>.vscode/**</exclude>
```
This should be added to https://creadur.apache.org/rat/apache-rat-plugin/rat-mojo.html, like `useEclipseDefaultExcludes` and others.

michael-o commented:
Ideally, this should land upstream: RAT would contain dot vscode by default. Until this is done, we should have it locally with a comment to remove when RAT is fixed.

jmhodges commented:
Thanks for looking at this!

I dug in and it looks like the answer is just to add the directory to `.gitignore` and rat does what you want. We'd want to do that `.gitignore` change, anyway, to prevent future problems from vscode-using contributors, so I've updated the PR to match

slawekjaranowski commented on maven-settings-builder/pom.xml:
```diff
@@ -72,8 +72,12 @@ under the License.
       <plugin>
         <groupId>org.eclipse.sisu</groupId>
         <artifactId>sisu-maven-plugin</artifactId>
+        <!-- <goals>
+          <goal>main-index</goal>
+          <goal>test-index</goal>
+        </goals> -->
```
What for is it?

jmhodges replied:
Sorry, a mistaken addition (while working on some red squiggly problems)

jmhodges commented:
(I deleted a comment I wrote because I hadn't seen all the commentary)

Thanks for the feedback!

Totally agreed on upstreaming this. I'm making a ticket for them

I do think adding to `.gitignore` is still useful. Adding to `.gitignore`  turns out to be the only change we need to convince rat to not alarm, makes sure reviewers aren't having to waste cycles ensuring that `.vscode` doesn't get accidentally added to git, and avoids contributors having to make global changes to their config to get a working build.

I've updated the PR to only touch `.gitignore`.

jmhodges commented:
Made the ticket! https://issues.apache.org/jira/browse/RAT-292

Let me know if you're okay with just the `.gitignore` change. Thanks again for reviewing so quickly!
